src/lightly_train/_commands/train_helpers.py

```diff
@@ -60,9 +60,15 @@ def get_transform_args(
     if transform_args is None:
         # We need to typeignore here because a MethodTransformArgs might not have
         # defaults for all fields, while its children do.
-        return transform_args_cls()  # type: ignore[call-arg]
+        transform_args = transform_args_cls()  # type: ignore[call-arg]
+    else:
+        transform_args = validate.pydantic_model_validate(
+            transform_args_cls, transform_args
+        )
 
-    return validate.pydantic_model_validate(transform_args_cls, transform_args)
+    transform_args.resolve_auto()
+    transform_args.resolve_incompatible()
```

Comment:
Had to add `resolve_auto` because `num_channels` is by default `"auto"` in the transforms. This is nice if you set a custom `normalize` or `channel_drop` augmentation because then the number of channels are automatically inferred.

Similarly, I had to add `resolve_incompatible` because there are now some situations where transforms don't work if `num_channels != 3`. Check `semantic_segmentation_transforms.py` for details.